File: app/src/main/java/com/example/androidmvvm/viewmodel/CounterViewModel.java
package com.example.androidmvvm.viewmodel;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.example.androidmvvm.model.Count;

/**
 * Created by tho nguyen on 2019-05-09.
 */
public class CounterViewModel extends ViewModel {

    private MutableLiveData<Count> counter = new MutableLiveData<>();
    private Count count = new Count();


    public void onButtonClick() {
        int value = count.getCount();
        count.setCount(++value);
        counter.setValue(count);
    }

    public MutableLiveData<Count> getCounter() {
        return counter;
    }

}

File: app/src/main/java/com/example/androidmvvm/viewmodel/CounterViewModel.kt
package com.example.androidmvvm.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

import com.example.androidmvvm.model.Count

/**
 * Created by tho nguyen on 2019-05-09.
 */
class CounterViewModel : ViewModel() {

    val counter = MutableLiveData<Count>().apply {
        value = Count()
    }


    fun onButtonClick() {
        counter.value = counter.value.apply { this!!.index++ }

    }

}
